ush/wdqms.py

```diff
@@ -0,0 +1,753 @@
+            u_df = pd.DataFrame(df_dict['u'])
+            u_df = u_df.rename({'Observation_Class': 'var_id',
+                                'u_Observation': 'Observation',
+                                'u_Obs_Minus_Forecast_adjusted': 'Obs_Minus_Forecast_adjusted'},
+                               axis=1)
+            u_df['var_id'] = self.wdqms_type_dict[self.wdqms_type]['variable_ids']['u']
+
+            v_df = pd.DataFrame(df_dict['v'])
+            v_df = v_df.rename({'Observation_Class': 'var_id',
+                                'v_Observation': 'Observation',
+                                'v_Obs_Minus_Forecast_adjusted': 'Obs_Minus_Forecast_adjusted'},
+                               axis=1)
+            v_df['var_id'] = self.wdqms_type_dict[self.wdqms_type]['variable_ids']['v']
+
+            df = pd.concat([u_df, v_df])
+
+        else:
+            for col in column_list:
+                data = self._grab_netcdf_data(file, col)
+                df_dict[col] = data
+
+            df = pd.DataFrame(df_dict)
+            df = df.rename({'Observation_Class': 'var_id'}, axis=1)
+            df['var_id'] = self.wdqms_type_dict[self.wdqms_type]['variable_ids'][variable]
+
+        # Add datetime column to dataframe
+        df['Datetime'] = datetime
+
+        # Subtract longitudes > 180 by 360 to be negative
+        df.loc[df['Longitude'] > 180, 'Longitude'] -= 360
+
+        logging.debug("Exiting read_gsi_diag()")
+
+        return df
+
+    def _grab_netcdf_data(self, file, var):
+        """
+        Opens and grabs data based on column name.
+        Args:
+            file : (str) netCDF GSI file
+            var  : (str) the variable to be extracted
+        Returns:
+            data : (array) values from the specified variable
+        """
+        logging.debug('Working in grab_netcdf_data()')
+
+        with Dataset(file, mode='r') as f:
+            # Station_ID and Observation_Class variables need
+            # to be converted from byte string to string
+            if var == 'Datetime':
+                data = f.date_time
+
+            elif var in ['Station_ID', 'Observation_Class']:
+                data = f.variables[var][:]
+                data = [i.tobytes(fill_value='        ', order='C')
+                        for i in data]
+                data = np.array(
+                    [''.join(i.decode('UTF-8', 'ignore').split())
+                     for i in data])
+
+            # Grab variables with only 'nobs' dimension
+            elif len(f.variables[var].shape) == 1:
+                data = f.variables[var][:]
+
+        logging.debug("Exiting grab_netcdf_data()")
+
+        return data
+
+    def _df_to_csv(self, df):
+        """
+        Produce output .csv file from dataframe.
+        """
+        logging.info("Working in df_to_csv()")
+        logging.info(f'Coverting dataframe to .csv file for {
+                     self.wdqms_type} data ...')
```

Comment:
These two lines create the following error message:
```
  File "/lfs/h2/emc/da/noscrub/emily.liu/git/global-workflow-wdqms/ush/wdqms.py", line 695
    logging.info(f'Coverting dataframe to .csv file for {
                                                        ^
SyntaxError: EOL while scanning string literal
13 + rc=1
```

Reply:
The global-workflow `gdaswdqms` completed with three CVS output successfully after the above EOL error fixed.

Reply:
Thanks @emilyhcliu for the testing.